<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KnowledgeClass</class>
 <widget class="QMainWindow" name="KnowledgeClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>561</width>
    <height>622</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>561</width>
    <height>622</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>561</width>
    <height>622</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Knowledge</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>70</y>
      <width>321</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Question information </string>
    </property>
    <widget class="QLineEdit" name="questionNumberLineEdit">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>50</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="frame">
      <bool>true</bool>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="questionNumberOutOfLineEdit">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>50</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>91</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Question Number:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>50</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Out of:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="questionTitleLineEdit">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>20</y>
       <width>201</width>
       <height>20</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>91</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Question title:</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>520</y>
      <width>542</width>
      <height>25</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="testCustomQuickSortPushButton">
       <property name="text">
        <string>Test Custom Quick Sort</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string>Test Custom Binary Search</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_2">
       <property name="text">
        <string>Manipulate 2D Array </string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>160</y>
      <width>321</width>
      <height>351</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>Question</string>
       </property>
       <widget class="QTextEdit" name="quesitonTextEdit">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>301</width>
          <height>141</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::Box</enum>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="title">
        <string>Answer</string>
       </property>
       <widget class="QTextEdit" name="answerTextEdit">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>301</width>
          <height>141</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::Box</enum>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>211</width>
      <height>501</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Quesiton Batch Selection</string>
       </property>
       <widget class="QLabel" name="label_4">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>81</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Search:</string>
        </property>
       </widget>
       <widget class="QListView" name="batchListView">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>80</y>
          <width>191</width>
          <height>161</height>
         </rect>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SingleSelection</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
       </widget>
       <widget class="QLineEdit" name="batchSearchLineEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>20</y>
          <width>101</width>
          <height>20</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="batchBSearchLineEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>50</y>
          <width>101</width>
          <height>20</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>50</y>
          <width>81</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Binary Search:</string>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="title">
        <string>Question Selection</string>
       </property>
       <widget class="QListView" name="questionListView">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>80</y>
          <width>191</width>
          <height>161</height>
         </rect>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
       </widget>
       <widget class="QLabel" name="label_5">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>81</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Search:</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="questionSearchLineEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>20</y>
          <width>101</width>
          <height>20</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="questionBSearchLineEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>50</y>
          <width>101</width>
          <height>20</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>50</y>
          <width>81</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Binary Search:</string>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>10</y>
      <width>321</width>
      <height>51</height>
     </rect>
    </property>
    <property name="title">
     <string>Quesiton Batch</string>
    </property>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>91</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Batch title:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="batchTitleLineEdit">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>20</y>
       <width>201</width>
       <height>20</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>550</y>
      <width>541</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Write Read from binary file using random-access algorithms</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>561</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="contextMenuPolicy">
    <enum>Qt::CustomContextMenu</enum>
   </property>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="knowledge.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>KnowledgeClass</receiver>
   <slot>on_actionExit_Triggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batchSearchLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>KnowledgeClass</receiver>
   <slot>on_batchSearchLineEdit_TextChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>227</x>
     <y>70</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>questionSearchLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>KnowledgeClass</receiver>
   <slot>on_quesitonSearchLineEdit_TextChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>211</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>278</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batchBSearchLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>KnowledgeClass</receiver>
   <slot>on_batchBSearchLineEdit_TextChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>182</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>223</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>questionBSearchLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>KnowledgeClass</receiver>
   <slot>on_quesitonBSearchLineEdit_TextChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>347</y>
    </hint>
    <hint type="destinationlabel">
     <x>228</x>
     <y>366</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>testCustomQuickSortPushButton</sender>
   <signal>clicked()</signal>
   <receiver>KnowledgeClass</receiver>
   <slot>on_testCustomQuickSortPushButton_Clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>564</y>
    </hint>
    <hint type="destinationlabel">
     <x>404</x>
     <y>465</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>KnowledgeClass</receiver>
   <slot>on_testCustomBinarySearchPushButton_Clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>237</x>
     <y>542</y>
    </hint>
    <hint type="destinationlabel">
     <x>457</x>
     <y>553</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>KnowledgeClass</receiver>
   <slot>on_manipulate2DArray_Clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>550</x>
     <y>564</y>
    </hint>
    <hint type="destinationlabel">
     <x>389</x>
     <y>572</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>KnowledgeClass</receiver>
   <slot>on_writeReadBinaryFile_Clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>126</x>
     <y>588</y>
    </hint>
    <hint type="destinationlabel">
     <x>153</x>
     <y>609</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_actionExit_Triggered()</slot>
  <slot>on_batchSortByComboBox_CurrentIndexChanged(int)</slot>
  <slot>on_questionSortByComboBox_CurrentIndexChanged(int)</slot>
  <slot>on_testCustomQuickSortPushButton_Clicked()</slot>
  <slot>on_nextPushButton_Clicked()</slot>
  <slot>on_batchListView_Clicked(QModelIndex)</slot>
  <slot>on_questionListView_Clicked(QModelIndex)</slot>
  <slot>on_batchSearchLineEdit_TextChanged(QString)</slot>
  <slot>on_quesitonSearchLineEdit_TextChanged(QString)</slot>
  <slot>on_batchListView_IndexesMoved(QModelIndexList)</slot>
  <slot>on_questionListView_IndexesMoved(QModelIndexList)</slot>
  <slot>on_batchListView_Activated(QModelIndex)</slot>
  <slot>on_questionListView_Activated(QModelIndex)</slot>
  <slot>on_batchBSearchLineEdit_TextChanged(QString)</slot>
  <slot>on_quesitonBSearchLineEdit_TextChanged(QString)</slot>
  <slot>on_testCustomBinarySearchPushButton_Clicked()</slot>
  <slot>on_manipulate2DArray_Clicked()</slot>
  <slot>on_writeReadBinaryFile_Clicked()</slot>
  <slot>on_randomAccessHandling_Clicked()</slot>
 </slots>
</ui>
